### PR TITLE
update go.mod

### PR DESCRIPTION
**Pull request by g-k**
refs: https://bugzilla.mozilla.org/show_bug.cgi?id=1477319 for autograph 1.0.2 deploy (I'll need to tag a new release and update the bug)

Changes:

* update dep versions
* add dep hashes in `go.sum`
* fix CI "go: error loading module requirements" errors per https://github.com/golang/go/issues/26594 these were due to an older git version 2.11
* add clang to CI and Dockerfile so `vgo build` run as part of `vgo install` passes

r? @milescrabill

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -19,8 +19,10 @@ jobs:
       - run:
           name: Install Packages
           command: |
+              echo 'deb http://ftp.debian.org/debian stretch-backports main' >> /etc/apt/sources.list
               apt update
-              apt install -y libltdl-dev netcat
+              apt install -y libltdl-dev netcat clang
+              apt-get -y -t stretch-backports install git
```
Comment (milescrabill) [inline .circleci/config.yml]: interesting that git isn't in the mainline repo

File: Dockerfile [hunk 2]
```diff
@@ -3,13 +3,14 @@ MAINTAINER Mozilla
 EXPOSE 8080
 
 RUN addgroup --gid 10001 app && \
-
+    # \
```
Comment (milescrabill) [inline Dockerfile]: remove this

File: Dockerfile [hunk 3]
```diff
@@ -3,13 +3,14 @@ MAINTAINER Mozilla
...
     adduser --gid 10001 --uid 10001 \
     --home /app --shell /sbin/nologin \
     --disabled-password app && \
-
+    # \
```
Comment (milescrabill) [inline Dockerfile]: and this. and maybe make the `apt` stuff a separate `RUN` step

File: .circleci/config.yml (hunk 1 above)
Comment (g-k) [reply .circleci/config.yml]: yeah 2.18 was just released in Jun. I think vgo is using Transfer protocol v2 for partial clones.

**Pull request merged**